edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/ServiceIdentityTree.cs

```diff
@@ -118,6 +118,27 @@ public async Task<Option<string>> GetAuthChain(string id)
             return Option.None<string>();
         }
 
+        public async Task<IList<ServiceIdentity>> GetImmediateChildren(string id)
+        {
+            Preconditions.CheckNonWhiteSpace(id, nameof(id));
+
+            var children = new List<ServiceIdentity>();
+
+            using (await this.nodesLock.LockAsync())
+            {
+                if (this.nodes.TryGetValue(id, out ServiceIdentityTreeNode treeNode))
+                {
+                    IList<ServiceIdentityTreeNode> childNodes = treeNode.GetAllChildren();
+                    foreach (ServiceIdentityTreeNode node in childNodes)
```

Comment:
use List AddRange method.
```suggestion
                    children.AddRange(treeNode.GetAllChildren());
```

Reply:
they're different types, but AddRange still works if I apply a Select

---
In reply to: [451895233](https://github.com/Azure/iotedge/pull/3149#discussion_r451895233) [](ancestors = 451895233)